libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/MapboxCarMapSurfaceUtils.kt

```diff
@@ -0,0 +1,32 @@
+@file:OptIn(MapboxExperimental::class)
+
+package com.mapbox.androidauto.car
+
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.Style
+import com.mapbox.maps.extension.androidauto.MapboxCarMapSurface
+import com.mapbox.maps.plugin.delegates.listeners.OnStyleLoadedListener
+
+fun MapboxCarMapSurface.handleStyleOnAttached(
+    block: (Style) -> Unit
+): OnStyleLoadedListener {
+    getStyle { block(it) }
+
+    return OnStyleLoadedListener {
+        getStyle { block(it) }
+    }.also {
+        mapSurface.getMapboxMap().addOnStyleLoadedListener(it)
+    }
+}
+
+fun MapboxCarMapSurface.handleStyleOnDetached(
+    listener: OnStyleLoadedListener?,
+    block: (Style) -> Unit
+) {
+    getStyle { block(it) }
+    listener?.let { mapSurface.getMapboxMap().removeOnStyleLoadedListener(it) }
+}
```

Comment:
```suggestion
fun MapboxCarMapSurface.handleStyleOnDetached(listener: OnStyleLoadedListener?): Style? {
    listener?.let { mapSurface.getMapboxMap().removeOnStyleLoadedListener(it) }
    return mapSurface.getMapboxMap().getStyle()
}
```

Reply:
why returning a style object is better than handling it with block fun?

Reply:
Less overhead due to anonymous class instantiation.

Reply:
it's not critical, let's keep it consistent with `handleStyleOnAttached`